### DISCUSSION TITLE
Realistic bone ultrasound simulation

### DISCUSSION THREAD

**Discussion by catarinalopes21** [PlusToolkit/PlusLib#1216] [General]
How do I make this ultrasound simulation of the bone more similar to a real bone surface ultrasound?

![388184318-28d32c75-4b60-4898-883b-c32edb421462](https://github.com/user-attachments/assets/177f3074-8105-497a-a146-2344e6e4a78e)

By this I mean having the fat/muscle-like image of the ultrasound in the place where the gel appears in the image. So something like this:

![388186589-5518c1d1-6418-4083-af66-c86fdc050159](https://github.com/user-attachments/assets/60f58f19-0ee6-42f4-b4d9-c9b9a00c27be)

**Comment by lassoan** [comment]
You can adjust the acoustic parameters of the objects in the Plus device set configuration file, but the noise pattern will not likely to get much more realistic than this. The ultrasound simulator does not aim for reproducing every aspect of an ultrasound image, but it creates an image with realistic shapes and overall brightness, to help trainees learn how geometry and overall appearance. This goal is achieved here (reflection from fat is bright, then there is a thin bright reflection from the bone surface, then it quickly fades to black).

If your trainees are very comfortable with interpreting these cleaner images then you can go to the next level and generate more complex images by training a GAN or other AI models.
